### PR TITLE
Case 19188: Potential 2D overlay + Tablet threading fixes

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/19188/click-on-PAL-crash

Test plan:
- Click the tablet buttons a lot.  Go to different domains.  Click em more.  You shouldn't crash.
- Notifications should still work.
- Run [this](https://gist.githubusercontent.com/SamGondelman/a339b6aa11336ea254936f26592685fa/raw/6b3a1ac5ce2c7b9a6666545d30510527a4a3580b/QMLThreadTest.js).  You'll see a text overlay flashing around your screen super quickly, and a flashing button on your toolbar (or tablet in HMD).  You shouldn't crash.
- I think we should merge this and then monitor the crashes in the ticket after 80 is released.

### CODE REVIEW

File: interface/src/ui/overlays/Overlays.cpp [hunk 1]
```diff
@@ -772,29 +772,29 @@ QUuid Overlays::addOverlay(const QString& type, const QVariant& properties) {
...
-        overlay->setProperties(properties.toMap());
-        return add2DOverlay(overlay);
+        Overlay::Pointer overlay;
+        if (type == ImageOverlay::TYPE) {
+            overlay = Overlay::Pointer(new ImageOverlay(), [](Overlay* ptr) { ptr->deleteLater(); });
+        } else if (type == TextOverlay::TYPE) {
+            overlay = Overlay::Pointer(new TextOverlay(), [](Overlay* ptr) { ptr->deleteLater(); });
+        } else if (type == RectangleOverlay::TYPE) {
+            overlay = Overlay::Pointer(new RectangleOverlay(), [](Overlay* ptr) { ptr->deleteLater(); });
+        }
```
Comment (jherico) [inline interface/src/ui/overlays/Overlays.cpp]: I would suggest an else here with an explicit exception, or a case statement on `type` with a `Q_UNREACHABLE` or something.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17769/HighFidelity-Beta-PR15066-b26b02f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17769/HighFidelity-Quest-Beta-PR15066-b26b02f.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17769/HighFidelity-Beta-PR15066-b26b02f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17769/HighFidelity-Beta-Interface-PR15066-b26b02f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17769/HighFidelity-Beta-PR15066-b26b02f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17769/HighFidelity-Beta-Interface-PR15066-b26b02f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17789/HighFidelity-Beta-PR15066-cfb8496.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17789/HighFidelity-Beta-Interface-PR15066-cfb8496.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17789/HighFidelity-Beta-PR15066-cfb8496.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17789/HighFidelity-Beta-Interface-PR15066-cfb8496.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17789/HighFidelity-Beta-PR15066-cfb8496.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17789/HighFidelity-Quest-Beta-PR15066-cfb8496.apk)

**Pull request merged**